### DISCUSSION TITLE
on_event usage

### DISCUSSION THREAD

**Discussion by iyakushev** [fastapi/fastapi#8569] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()

@app.on_event("startup")  # this produces type error.
def foo():
  ...
```

### Description

The code example above produces type errors while using `strict` mode on pyright and pylance as a consequence. Example errors: 
```
Untyped function decorator obscures type of function; ignoring decorator
Type of "on_event" is partially unknown. Type of "on_event" is "(event_type: str) -> ((...) -> Unknown)"
```

The problem lies in the Starlette method annotation of return value; it is untyped:
```python
    # starlette/applications.py
    ...
     
    # The following usages are now discouraged in favour of configuration
    #  during Starlette.__init__(...)
    def on_event(self, event_type: str) -> typing.Callable:
        return self.router.on_event(event_type)
     
    # ========= Possible quick fix
    # Alternatively, providing a broad type of return value removes complaints
    def on_event(self, event_type: str) -> typing.Callable[..., typing.Any]:
        return self.router.on_event(event_type)  # NB on_event uses untyped Callables as well
```
However, I'm not sure that this needs a PR since Starlette supports `on_startup` argument. 
If this shouldn't be fixed (due to mypy being fine with it, for example), perhaps it is better to discourage this in the docs and show usage of `FastAPI(on_startup=[...])`. At the moment [docs suggest](https://fastapi.tiangolo.com/advanced/events/#startup-event) to use `@app.on_event(...)` 

### Operating System

Linux

### FastAPI Version

0.73.0

### Python Version

3.10.1

**Comment by ArthurKoba** [comment]
### I confirm that there is a problem. 

**Operating System**: _Windows 10_
**Python Version**: _3.9.1_
**FastAPI Version**: _0.75.1_
**Starlette Version**: _0.17.1_

**Comment by doublethink13** [comment]
Following
